application/apps/indexer/session/src/state/session_file.rs

```diff
@@ -46,13 +46,21 @@ pub enum SessionFileState {
     NoChanges,
 }
 
+#[derive(Debug, Clone)]
+pub enum SessionFileStage {
+    NotCreated,
+    Linked,
+    NotLinked,
+}
+
```

Comment:
one possibility. but makes matching a little complex.
other possibility could be:
``` rust
#[derive(Debug)]
pub struct SessionFile {
    pub grabber: Option<Box<Grabber>>,
    pub filename: Option<PathBuf>,
    pub writer: Option<BufWriter<File>>,
    pub last_message_timestamp: Instant,
    pub sources: SourceIDs,
    pub origin: Option<SessionFileOrigin>,
}
...
pub enum SessionFileOrigin {
    TextFile(PathBuf),
    Streams(streams),
}
```

Reply:
looks good

Reply:
yes, make sense... I've updated it